### PR TITLE
Submission Stats

### PR DESCRIPTION
**Pull request by knockknockyoo**
1. Node Types
 - new, passed, warning, error
2. GraphQL Query: submissionStats
3. DataRecordService
- submissionStats
4. User Permission for submissionStats

### CODE REVIEW

**Review by vshand11 — CHANGES_REQUESTED** [review]
Hi Young, found one issue need code change so far.   Please solve the conflicts so I can test in local.

File: services/data-record-service.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        this.dataRecordsCollection = dataRecordsCollection;
+    }
+
+    async submissionStats(submissionID) {
+        const groupPipeline = { "$group": { _id: "$nodeType", count: { $sum: 1 }} };
+        const groupByNodeType = await this.dataRecordsCollection.aggregate([{ "$match": {submissionID: submissionID, status: {$in: [NODE_TYPE.NEW, NODE_TYPE.PASSED, NODE_TYPE.WARNING, NODE_TYPE.ERROR]}}}, groupPipeline]);
+
+        const statusPipeline = { "$group": { _id: "$status", count: { $sum: 1 }} };
+        const promises = groupByNodeType.map(async node =>
+            [await this.dataRecordsCollection.aggregate([{ "$match": {submissionID: submissionID, nodeType: node?._id}}, statusPipeline]), node?._id]
```
[the remark this answers is not in the document]
Comment (knockknockyoo) [reply services/data-record-service.js:21]: We used aggregate groupby in MonogoDB, and the result is expected to be _id per element.

**Review by vshand11 — APPROVED** [review]
New version works and passed my testing in local.

File: services/data-record-service.js [hunk 2]
```diff
@@ -96,6 +120,59 @@ class Message {
...
+        this.passed = passedCount;
+        this.warning= warningCount;
+        this.error = errorCount;
+    }
+
+    static createStat(nodeName) {
+        return new Stat(nodeName, 0,0,0,0, 0);
+    }
+
+    #addTotal(total) {
```
Comment (AustinSMueller) [inline services/data-record-service.js:137]: I think we should remove the "#" from the start of this function name. I don't know if there is any issue with it but I'd rather keep the function names alphanumeric to match the existing code style. 

If there is a special function for this naming convention please let me know

**Review by AustinSMueller — CHANGES_REQUESTED** [review]

File: constants/submission-constants.js [hunk 3]
```diff
@@ -22,6 +22,12 @@ module.exports = Object.freeze({
         REJECT_SUBMIT: "Reject_Submitted",
         REJECT_RELEASE: "Reject_Released"
     },
+    NODE_TYPE: {
```
[the remark this answers is not in the document]
Comment (knockknockyoo) [reply constants/submission-constants.js]: updated the constant name

File: services/data-record-service.js (hunk 2 above)
Comment (knockknockyoo) [reply services/data-record-service.js:137]: It is the purpose of private method

File: services/submission.js [hunk 4]
```diff
@@ -680,6 +692,24 @@ function validateListSubmissionsParams (params) {
...
+        return;
+    }
+    if (userRole === USER.ROLES.ORG_OWNER && userInfo?.organization?.orgID === aSubmission?.organization?._id) {
+        return;
+    }
+    if (userRole === USER.ROLES.SUBMITTER && userInfo?._id === aSubmission?.submitterID) {
+        return;
+    }
+    if (userRole === USER.ROLES.DC_POC && userInfo?.dataCommons.includes(aSubmission?.dataCommons)) {
+        return;
```
[the remark this answers is not in the document]
Comment (knockknockyoo) [reply services/submission.js:703]: updated

**Review by AustinSMueller — APPROVED** [review]

**Pull request merged**